File: example/src/main/java/net/droidlabs/viking/example/MvvmFragment.java
package net.droidlabs.viking.example;

import net.tango.viking.annotations.AutoModule;
import net.droidlabs.viking.example.R;
import net.droidlabs.viking.example.databinding.FragmentTestBinding;
import net.droidlabs.viking.mvvm.FragmentView;

@AutoModule(scopes = {MapActivity.class})
public class MvvmFragment extends FragmentView<TestViewModel2, FragmentTestBinding> {

  public static MvvmFragment newInstance() {
    return new MvvmFragment();
  }

  public MvvmFragment() {
    super(R.layout.fragment_test, TestViewModel2.class);
  }

  @Override
  protected void bind(FragmentTestBinding binding) {

  }
}

File: example/src/main/java/net/droidlabs/viking/example/MvvmFragment.kt
package net.droidlabs.viking.example

import net.droidlabs.viking.annotations.AutoModule
import net.droidlabs.viking.example.MapActivity
import net.droidlabs.viking.example.databinding.FragmentTestBinding
import net.droidlabs.viking.mvvm.FragmentView

@AutoModule(scopes = [MapActivity::class])
class MvvmFragment : FragmentView<TestViewModel2, FragmentTestBinding>(
    layoutIdRes = R.layout.fragment_test,
    viewModelClass = TestViewModel2::class.java) {
  override fun bind(binding: FragmentTestBinding) {}

  companion object {
    fun newInstance(): MvvmFragment {
      return MvvmFragment()
    }
  }
}